packages/pyright-internal/src/languageService/signatureHelpProvider.ts

```diff
@@ -245,21 +249,29 @@ export class SignatureHelpProvider {
                 paramName = params[params.length - 1].name || '';
             }
 
-            parameters.push({
-                startOffset: label.length,
-                endOffset: label.length + paramString.length,
-                text: paramString,
-            });
+            const isKeywordOnly =
```

Comment:
I don't think this code works for a number of edge cases. It's mixing actual parameters (those that are in the `FunctionType.shared.parameters` array) with virtual parameters (those that are returned by `getParamListDetails`). I can say from personal experience that these should never be intermixed. The indexes of the actual parameters are not the same as the indexes of the virtual parameters. The `getParamListDetails` function does things like remove `/` and `*` separators, expand unpacked tuples for `*args`, and expand unpacked TypedDicts for `**kwargs`. So the list of virtual parameters can be quite different from the list of actual parameters. Prior to this PR, this function focused completely on actual parameters, so it was internally consistent. With this proposed change, there are cases that won't be handled correctly.

As a potential fix, you could call `getParamListDetails` and then search the list of virtual parameters for one whose "kind" is keyword-only and whose name matches that of the actual parameter. That way, you won't be depending on the indexes of the two lists being the same.

Reply:
Thanks for catching that. Does this look right now?